src/prog/thermo.f90

```diff
@@ -164,8 +174,75 @@ subroutine preigf(unit, freq, scale)
 end subroutine preigf
 
 
+! Remove the frequency shift caused by the bias RMSD employed in the SPH calculation:
+subroutine rescale_freq(env,argParser,mol, hessian, freq)
+    real(wp), allocatable, intent(inout) :: freq(:), hessian(:, :)
+    real(wp), allocatable :: freq_sph(:), hessian_sph(:, :) ! biased hessian from SPH
+    real(wp), allocatable :: htb(:,:), v(:), fc_tmp(:), fc_tb(:), fc_bias(:), freq_scal(:)
+    real(wp) :: alp1, alp2 ! factors for SPH scaling
+    integer :: i,j,n3
+    type(TEnvironment), intent(inout) :: env
+	!> Commandline argument parser
+	type(TArgParser), intent(inout) :: argParser
+	type(TReader) :: reader
+	type(TMolecule) :: mol
+	character(len=:), allocatable :: file
+	integer :: hFormat
+    logical :: ex
+
+   n3= 3*mol%n
+	hFormat = 1 ! hessian_sph must stem from xtb SPH calculation!
+	call argParser%nextFile(file)
+	write(env%unit,'(a)') 'Reading hessian_sph file: '//file
+   write(env%unit,'(a)') 'Removing the frequency shift caused by the bias RMSD:'
+	allocate(freq_sph(n3))
+	if (allocated(file)) then
+	   allocate(hessian_sph(n3,n3))
+	   call reader%open(file)
+	   call readHessian(env, mol, hessian_sph, reader, format=hFormat)
+	   call reader%close
+      call massWeightHessian(hessian_sph, atmass) ! keep units similar to hessian.f90 routines in main program
+      call massWeightHessian(hessian, atmass) ! keep units similar to hessian.f90 routines in main program
+      htb=hessian-hessian_sph
+
+      call diagHessian(env, hessian, freq)
+
+      allocate(v(n3),fc_tmp(n3),freq_scal(n3),fc_tb(n3),fc_bias(n3))
+      ! calculate fc_tb and fc_bias (same procedure as in hessian.f90)
+      alp1=1.27_wp
+      alp2=1.5d-4
```

Comment:
Can we avoid using fixed float point numbers?

Reply:
Or is the whole function "copied" from the `bhess` workflow?

Reply:
I just "copied" the function from the bhess workflow. Otherwise the bhess subroutine has to be also rewritten to avoid this "double code".

Reply:
Can't we just use it?

Reply:
Can you highlight which lines of code from `src/hessian.f90` are actually relevant and could be "recycled" here? So that we can think about putting these parts of the code into single functions that could be reused in this project. I'd strongly speak against implementing doubled code using fixed float point parameters (which is in itself already a bit of bad practice), if we can avoid it.

Reply:
It looks like it is mainly lines 397-416 from the updated hessian.F90 that could be reused here fairly easily.